src/transformers/models/bert/modeling_bert.py

```diff
@@ -669,12 +711,16 @@ def forward(
         if output_hidden_states:
             all_hidden_states = all_hidden_states + (hidden_states,)
 
+        next_cache = next_decoder_cache if use_cache else None
+        if return_legacy_cache:
+            next_cache = past_key_values.to_legacy_cache()
+
         if not return_dict:
             return tuple(
                 v
                 for v in [
                     hidden_states,
-                    next_decoder_cache,
+                    next_cache,
```

Comment:
Similarly here, we don't have any notion of current/next cache anymore:
```suggestion
                    past_key_values if not return_legacy_cache else past_key_values.to_legacy_cache()
```

Reply:
`if not return_dict:` can also be removed with check_model_inputs and `can_return_tuples`